File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/generate/OpenApiWalkerListener.java
package io.github.potjerodekool.openapi.common.generate;

import io.swagger.models.HttpMethod;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.Operation;
import io.swagger.v3.oas.models.media.Content;
import io.swagger.v3.oas.models.media.Schema;

public interface OpenApiWalkerListener {

    default void visitOperation(OpenAPI api,
                                HttpMethod method,
                                String path,
                                Operation operation) {
    }

    default void visitContent(OpenAPI openAPI,
                              HttpMethod httpMethod,
                              String path,
                              Operation operation, final Content content) {
    }

    default void visitSchema(OpenAPI openAPI,
                             HttpMethod httpMethod,
                             String path,
                             Operation operation,
                             Schema<?> schema,
                             String schemaName) {
    }

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/generate/OpenApiWalkerListener.kt
package io.github.potjerodekool.openapi.common.generate

import io.github.potjerodekool.openapi.common.ApiConfiguration
import io.swagger.models.HttpMethod
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.Operation
import io.swagger.v3.oas.models.media.Content
import io.swagger.v3.oas.models.media.Schema

interface OpenApiWalkerListener {

    fun visitOperation(
        api: OpenAPI,
        method: HttpMethod,
        path: String,
        operation: Operation?,
        apiConfiguration: ApiConfiguration
    ) {
    }

    fun visitContent(
        openAPI: OpenAPI,
        httpMethod: HttpMethod?,
        path: String?,
        operation: Operation?, content: Content?,
        apiConfiguration: ApiConfiguration
    ) {
    }

    fun visitSchema(
        openAPI: OpenAPI,
        httpMethod: HttpMethod?,
        path: String?,
        operation: Operation?,
        schema: Schema<*>?,
        contentType: ContentType?,
        schemaName: String?,
        apiConfiguration: ApiConfiguration
    ) {
    }
}
